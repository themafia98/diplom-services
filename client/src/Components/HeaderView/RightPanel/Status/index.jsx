import React from "react";
import { Icon } from "antd";

const Status = ({ statusApp, shouldUpdate }) => {
    return (
        <div className="statusApp">
            {shouldUpdate ? <Icon type="loading" /> : null}
            <div className={["statusAppIcon", statusApp].join(" ")}></div>
        </div>
    );
};

export default Status;
