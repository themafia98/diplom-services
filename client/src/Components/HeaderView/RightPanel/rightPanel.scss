.rightPanel {
  position: relative;
  overflow: hidden;
  height: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
  min-width: 300px;
  max-width: 500px;
  font-weight: bold;
  .userPopup {
    display: flex;
    align-items: center;
    height: 100%;
    margin-right: 30px;
    .userPopupMain {
      cursor: pointer;
      display: inline-flex;
      align-items: center;
      height: 100%;
      & > * {
        cursor: pointer;
      }
      & > a {
        margin-left: 5px;
      }
    }
  }
  .groupControllers {
    display: flex;
    align-items: center;
    & > * {
      padding: 5px;
    }
    .alertBell {
      cursor: pointer;
      font-size: 25px;
    }
  }
  .logout:hover {
    cursor: pointer;
    color: $important_default;
  }
}
