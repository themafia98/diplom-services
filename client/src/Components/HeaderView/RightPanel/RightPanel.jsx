import React from 'react';
import { rightPanelType } from '../types';
import Updater from '../../Updater';
import UserPopup from '../../UserPopup';

import NotificationPopup from '../../NotificationPopup';
import Status from './Status';

const RightPanel = props => {
  const { onUpdate, onLogout, goCabinet, status, shouldUpdate, udata = {}, notificationDep = {} } = props;

  return (
    <div className="headerControllers rightPanel">
      <UserPopup udata={udata} goCabinet={goCabinet} />
      <div className="groupControllers">
        <NotificationPopup udata={udata} notificationDep={notificationDep} />
        <Updater onClick={onUpdate} additionalClassName="updaterDefault" />
        <div onClick={onLogout} className="logout">
          Выйти
        </div>
        <Status shouldUpdate={shouldUpdate} statusApp={status} />
      </div>
    </div>
  );
};

RightPanel.defaultProps = {
  active: true,
};
RightPanel.propTypes = rightPanelType;

export default RightPanel;
