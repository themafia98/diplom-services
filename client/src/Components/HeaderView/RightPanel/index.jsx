import React from "react";
import Updater from "../../Updater";
import UserPopup from "../../UserPopup";

import Scrollbars from "react-custom-scrollbars";
import NotificationPopup from "../../NotificationPopup";
import Status from "./Status";

const RightPanel = ({ onUpdate, onLogout, goCabinet, status, shouldUpdate }) => {
    return (
        <div className="headerControllers rightPanel">
            <UserPopup goCabinet={goCabinet} />
            <div className="groupControllers">
                <NotificationPopup />
                <Updater onClick={onUpdate} additionalClassName="updaterDefault" />
                <div onClick={onLogout} className="logout">
                    Выйти
                </div>
                <Status shouldUpdate={shouldUpdate} statusApp={status} />
            </div>
        </div>
    );
};
export default RightPanel;
