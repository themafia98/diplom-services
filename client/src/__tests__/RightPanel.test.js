import React from "react";
import RightPanel from "../Components/HeaderView/RightPanel";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

describe("<RightPanel /> template", () => {
    test("should be render", () => {
        const props = {
            onUpdate: () => {},
            onClick: () => {}
        };
        const RightPanelWrapper = shallow(<RightPanel {...props} />);
        expect(RightPanelWrapper.find("Updater").prop("additionalClassName")).toEqual("updaterDefault");
        expect(toJson(RightPanelWrapper)).toMatchSnapshot();

        expect(RightPanelWrapper.find(".logout").simulate("click", [])).toBeTruthy();
    });
});
