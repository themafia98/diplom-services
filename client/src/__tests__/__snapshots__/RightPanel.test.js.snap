// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RightPanel /> template should be render 1`] = `
<div
  className="headerControllers rightPanel"
>
  <UserPopup
    goCabinet={null}
    statusApp=""
    udata={Object {}}
  />
  <div
    className="groupControllers"
  >
    <NotificationPopup
      notificationDep={Object {}}
      statusApp=""
      type="private"
      udata={Object {}}
    />
    <Updater
      additionalClassName="updaterDefault"
      className=""
      onClick={[Function]}
      statusApp=""
    />
    <div
      className="logout"
      onClick={null}
    >
      Выйти
    </div>
    <Status
      shouldUpdate={false}
      statusApp=""
    />
  </div>
</div>
`;
